main/design_documents/ARM_NEON_SVE_BRIDGE.rst

```diff
@@ -0,0 +1,44 @@
+=======================================
+Design Document for ARM_NEON_SVE_BRIDGE
+=======================================
+
+This document describes the design of the intrinsics defined in the ("NEON_SVE bridge")
+<../acle.rst#_ssec-NEONSVEBridge>. These intrinics add a
+convienient way to use NEON and SVE vectors interchangeably in C code with
+no performance overhead.
+
+Location
+########
+
+The intrinsics will be in a new header file called ``arm_extensions_bridge.h``.
+
+They will be conditionally defined with
+``#define __ARM_NEON_SVE_BRIDGE (__ARM_FEATURE_NEON && __ARM_FEATURE_SVE)``.
+``__ARM_FEATURE_NEON`` is used as it is aarch64 specific, and SVE is not
+supported in the aarch32 state.
+
+Including this header file will define ``__ARM_NEON_SVE_BRIDGE`` as 1.
+
+Suggestions & Considerations
+############################
+
+Why not enable the extraction of any 128-bit subvector in an SVE vector?
+------------------------------------------------------------------------
+
+Extracting other 128-bit subvectors would go against the no-op design.
+There are currently other methods of getting data this way from SVE
+subvectors using SVE intrinsics.
+
+Why not include 64-bit Vectors?
+-------------------------------
+
+64-bit vectors will not initially be added. The naming scheme of the intrinsics
+must respect possible future 64-bit additions. Following the NEON naming 
+convention, e.g. ``svget_neonq[_u16]`` and ``svset_neonq[_u16]`` resolves this.
```

Comment:
I believe the `q` part of the name could also be inferred from the argument even in the presence of a future 64-bit vector extension? Don't have a strong preference for making it implicit or not, just pointing out that it is possible.

Reply:
I'm updating the spec to expand on this. Basically C generics can't infer 64/128 bits from the return type alone (only the parameters) so the function name needs to reflect this